### DISCUSSION TITLE
Speed Insights "Awaiting Data"

### DISCUSSION THREAD

**Discussion by austintoddj** [vercel/community#3144] [Help] [answered]
### Summary

I'm currently on the Hobby plan and I've got a project using Vue/Vite. I'm not using a CSP, and I've enabled Speed Insights. I've gone through the docs a few times now and I'm still lost on why I'm not capturing any data.

<img width="1238" alt="speed-insights" src="https://github.com/vercel/community/assets/8037011/a81b6fbd-26a3-43b2-a744-f6fd04977705">

I've deployed and re-deployed the project. Any help here would be greatly appreciated 🙂 

### Steps to Reproduce

Not really much to replicate, other than me going in and enabling the feature.

**Comment by Xheldon** [comment]
I also encountered the same issue. Several days ago, I enabled Speed Insights, but it didn't show the results after 1 minute. So I thought maybe I needed to wait until the next day. However, when I remembered a few days later, I found that it was still on the "waiting for data" page.

By the way, my project is pure HTML and it is packaged by GitHub Action and then pushed to another repository to trigger automatic deployment on Vercel.

**Reply by Xheldon** [reply]
After consulting the documentation, I finally understood that Speed Insights is only applicable to frameworks like Next.js and not suitable for my static HTML. The latter requires the use of Speed Insights API.

**Reply by austintoddj** [reply]
I couldn't find that explicitly called out in the documentation, do you have a link to it?

**Reply by Thai-v4** [reply, marked as answer]
You can absolutly find this in the documentation, there is a link 
Go to the end of the page and you can see which framwork is supported. 

You can also use the Speed insight API if you project don't use the supported framework

**Reply by Xheldon** [reply]
FYI:
![CleanShot_2023_07_26_at_10_17_29@2x](https://github.com/vercel/community/assets/9780154/9608fd17-73b7-40bd-b4e1-a67d8e503a2b)

**Comment by austintoddj** [comment]
Like @No1ceTea suggested, using the [Speed Insights API](https://vercel.com/docs/concepts/speed-insights/api) worked.
